workflow/scripts/resources.py

```
@@ -63,41 +68,54 @@ def __init__(self, genome, build):
+
         if genome.lower() == "human":
             # create URLs for genome files
-            av = human_assembly_version[build]
-            
-            self.fa_url = f"{base_url_gencode}human/release_{build}/GRCh38.{av}.genome.fa.gz"
+            try:
+                av = human_assembly_version[build]
+            except KeyError:
+                raise ValueError(
+                    f"Invalid Gencode build number {build} for human genome. Please select a valid build number from {human_assembly_version.keys()}."
+                )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)